structs.go

```diff
@@ -132,7 +133,13 @@ func (s *Struct) FillMap(out map[string]interface{}) {
 			continue
 		}
 
-		out[name] = finalVal
+		if isSubStruct && (tagOpts.Has("flatten")) {
+			for k := range finalVal.(map[string]interface{}) {
+				out[k] = finalVal.(map[string]interface{})[k]
+			}
+		} else {
+			out[name] = finalVal
+		}
```

Comment:
@n3wtron can you please also update the doc comments above in `Map()`, so people know they can use it